chain/client/tests/process_blocks.rs

```diff
@@ -3182,3 +3184,96 @@ mod cap_max_gas_price_tests {
         assert!(!does_gas_price_exceed_limit(ProtocolFeature::CapMaxGasPrice.protocol_version()));
     }
 }
+
+#[cfg(test)]
+mod contract_precompilation_tests {
+    use super::*;
+    use near_primitives::contract::ContractCode;
+    use near_primitives::types::CompiledContractCache;
+    use near_store::{Store, StoreCompiledContractCache};
+    use near_vm_runner::{get_contract_cache_key, VMKind};
+
+    /// Deploys test contract to the first client, perform state sync for the second client and
+    /// check existence of contract in both caches.
+    #[test]
+    fn test_caching() {
```

Comment:
Please also check that the client that does state sync can also invoke the contract (process a function call transaction) without any issues